### PR TITLE
overwrite default Options from SuperFlag string

### PR DESCRIPTION
**Pull request by karlmcguire**
(This relates to DGRAPH-3003 and is blocked by dgraph-io/ristretto#247.)

1. I don't think there's a good way to include slices in a SuperFlag, so in `overwriteOptions` I only look at scalar, non-interface fields. I figured people could just use `WithLogger` and `WithEncryptionKey` instead of trying to accommodate it in SuperFlag. 
2. How should we handle `Dir` and `ValueDir` which previously [didn't really have a default value](https://github.com/dgraph-io/badger/blob/master/options.go#L124-L125)?

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1663)

### CODE REVIEW

**Review by manishrjain — APPROVED** [review]
:lgtm: Got a comment about adding a test with a legit value for super flag.

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1663#-:-MSnlqwe2vbn3GWevgTt:b6dxyd9)* status: all files reviewed, 1 unresolved discussion (waiting on @jarifibrahim and @karlmcguire)

___
*[options_test.go, line 12 at r1](https://reviewable.io/reviews/dgraph-io/badger/1663#-MSnlll78Tz5PXjFlkt-:-MSnlll78Tz5PXjFlkt0:bc0lq6m) ([raw file](https://github.com/dgraph-io/badger/blob/de940d2fab226ede66b6cc61bd06bdeae92e68a1/options_test.go#L12)):*
> ```Go
> func TestOptions(t *testing.T) {
> 	// copy all the default options over to a big SuperFlag string
> 	defaultSuperFlag := generateSuperFlag(DefaultOptions(""))
> ```

would be good to try to overwrite some option, to capture the most common use case.

**Review by manishrjain — APPROVED** [review]
@aman-bansal -- Can you resolve this and get it merged today?

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1663#-:-MT6BMgOAlndemEeMEmZ:bnk5gbp)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @jarifibrahim, @karlmcguire, and @manishrjain)

___
*[options.go, line 192 at r2](https://reviewable.io/reviews/dgraph-io/badger/1663#-MT6B9N9CHnYqEw2vOJo:-MT6B9NA0VvANAPdjekI:b-ylb7zo) ([raw file](https://github.com/dgraph-io/badger/blob/af72c79927a18a878a62f94665ada7e63214f0f7/options.go#L192)):*
> ```Go
> //
> // Unsupported: Options.Logger, Options.EncryptionKey
> func SetSuperFlag(superflag string, options Options) Options {
> ```

```
(Options) WithSuperFlag(superflag string) Options {
 ...
}
```
___
*[options_test.go, line 1 at r2](https://reviewable.io/reviews/dgraph-io/badger/1663#-MT6BH2c9rwW9BPbBCc9:-MT6BH2c9rwW9BPbBCcA:b-co4sw6) ([raw file](https://github.com/dgraph-io/badger/blob/af72c79927a18a878a62f94665ada7e63214f0f7/options_test.go#L1)):*
> ```Go
> package badger
> ```

license here.

File: options.go [hunk 1]
```diff
@@ -17,7 +17,10 @@
 package badger
 
 import (
+	"github.com/dgraph-io/ristretto/z"
```
Comment (jarifibrahim) [inline options.go:20]: run `gofmt` please

File: options_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package badger
```
Comment (jarifibrahim) [inline options_test.go:1]: license missing

**Pull request merged**